app/services/flow/base_flow.rb

```diff
@@ -23,6 +24,11 @@ def redirect_to(url)
       @redirect = url
     end
 
+    def render_json(json, status: nil)
+      @json = json
+      @http_status = status || :ok
+    end
```

Comment:
do we expect a lot more steps to render JSON like this? previously we had been making our JSON endpoints out-of-band from the flows, like the api image uploads controller, this seems like a lot to add for one step?

Reply:
The separate controller had a ton of extra code in it to make it satisfy the steps and call all the routines you get for free inside the state machine.

Reply:
Ok! But the first question: Do we expect a lot more steps to render JSON like this?

Reply:
It's not a step it's an action and can be called without any dependence on ordering.  So I would expect it could be the template for future APIs where you need to modify the state of the machine depending on the results of an API call.  And since an action doesn't typically render an erb it doesn't hurt to make it easy to return JSON.

Reply:
https://github.com/18F/identity-idp/blob/d22a63ae3df549bd066bc30354d27798d33a691c/app/services/idv/flows/doc_auth_flow.rb#L25